api/topology/src/main/java/quarks/topology/services/ApplicationService.java

```diff
@@ -63,4 +65,11 @@ Licensed to the Apache Software Foundation (ASF) under one
      * @see ApplicationServiceMXBean
      */
     void registerTopology(String applicationName, BiConsumer<Topology, JsonObject> builder);
+    
```

Comment:
Ah, I see how a register-after-submit wouldn't work for that case.  I was basing my comment/question on the just the "monitoring" and "restart failed jobs" use case in the title, etc.